### DISCUSSION TITLE
specify layout globally, so pages don't have to do it in frontmatter

### DISCUSSION THREAD

**Discussion by ahillio** [11ty/eleventy#2056] [Q&A] [answered]
I'd like to have a standard liquid layout be applied to all my markdown content pages.  I don't want to have to specify `layout: the-same-layout.liquid` in every markdown file.  I had hoped I could just have a global data file specify `"layout": "the-same-layout.liquid",` one single time so every one of my pages doesn't have to do that in its frontmatter, but I don't seem to be able to.  I saw a starterkit that had a global/default layout template that was automatically used for all content pages, but I wasn't sure how to replicate that.  Is there an easy/standard way to achieve this?

**Comment by pdehaan** [comment, marked as answer]
@ahillio It might depend on how your site is set up. If you have all your *.md files in a subdirectory, you could try using a [directory data file](https://www.11ty.dev/docs/data-template-dir/) which sets `"layout": "the-same-layout.liquid"` via JSON or JavaScript.

If you aren't using a subfolder, you could possibly try creating a _data/layout.json file and paste the following. It worked on my simple test case, but it was admittedly a very basic test case with only 3 templates and I didn't try overwriting the layout in any of other templates, so your mileage may vary:

```json
"layouts/the-same-layout.liquid"
```

**Reply by ahillio** [reply]
Is `layout.json` in the docs at all?  I haven't seen it there.  And I wonder what other options there are for particular files in the `data` dir.

I put `"layout.liquid"` (without "layouts/") in this `_data/layout.json` and that seems to be just what I was looking for.

**Reply by pdehaan** [reply]
I don't think it's documented. Just something I stumbled upon while playing w/ 11ty one day and realized it's just adding a global variable using a global data file while debugging some stuff w/ global "eleventyComputed.js" data files.

It seemed to work [at the time] for my simple use case, but not something I'd use a lot of since it'd clog up the global data file directory with a bunch of one line files, and I don't think I did enough experimentation or debugging around how easy it is to override the global data file with directory data files, template data files, or just frontmatter (with and without using eleventyComputed).

**Reply by cfjedimaster** [reply]
Id say it's not documented, but implied. I'd be in favor of doing a PR to document it if we agree it's *intended.*.

**Reply by pdehaan** [reply]
OK, now this is starting to come back to me. Here's a few existing conversations on the same topic (see https://github.com/11ty/eleventy/issues/380#issuecomment-568033456 for a similar solution as above):
- https://github.com/11ty/eleventy/issues/380
- https://github.com/11ty/eleventy/issues/245

I think I wasn't really excited to file an issue for more documentation (in 11ty-website repo) because it felt like a [hopefully temporary] workaround for #245. If we had top-level directory data file support, you wouldn't necessarily need to put specially named files in the _data/* directory.

And a case where using this "trick" might not always work like you expect:
- https://github.com/11ty/eleventy/issues/1791
